addons/l10n_it_xml_export/models/tax_report.py

```diff
@@ -0,0 +1,120 @@
+            'type': 'ir.actions.act_window',
+            'res_id': new_wizard.id,
+            'target': 'new',
+            'context': dict(self._context, l10n_it_xml_export_monthly_tax_report_options=options),
+        }
+
+    def export_tax_report_to_xml(self, options):
+        xml_content = self.env["ir.qweb"]._render("l10n_it_xml_export.tax_report_export_template", self._get_xml_export_data(options))
+        xml_content = Markup("""<?xml version="1.0" encoding="UTF-8" standalone="yes"?>""") + xml_content
+        xml_content = xml_content.encode()
+
+        with file_open("l10n_it_xml_export/data/validation/fornituraIvp_2018_v1.xsd", 'rb') as xsd:
+            xsd_schema = etree.XMLSchema(etree.parse(xsd, parser=etree.XMLParser()))
+            try:
+                xsd_schema.assertValid(etree.fromstring(xml_content))
+            except etree.DocumentInvalid as xml_errors:
+                self.env['bus.bus']._sendone(
+                    self.env.user.partner_id,
+                    'simple_notification',
+                    {
+                        'type': 'warning',
+                        'title': _('XML Validation Error'),
+                        'message': _("Some values will not pass the authority's validation, please check them before submitting your file: %s" % [error.path.split(":")[-1] for error in xml_errors.error_log]),
+                        'sticky': True,
+                    },
+                )
+
+        return {
+            "file_name": self.env["account.report"].browse(options["report_id"]).get_default_report_filename(options, 'xml'),
+            "file_content": xml_content,
+            "file_type": "xml",
+        }
+
+    def _get_xml_export_data(self, options):
+        options_date_to = fields.Date.from_string(options["date"]["date_to"])
+        report = self.env["account.report"].browse(options["report_id"])
+        company = report._get_sender_company_for_export(options)
+        report_lines = report._get_lines(options)
+        colname_to_idx = {col['expression_label']: idx for idx, col in enumerate(options.get('columns', []))}
+        lines_mapping = {
+            line['columns'][colname_to_idx['balance']]['report_line_id']: "{:.2f}".format(float(line['columns'][colname_to_idx['balance']]['no_format'])).replace(".", ",") if line['columns'][colname_to_idx['balance']]['no_format'] else False for line in report_lines
+        }
+        report_lines_data = {}
+        for record in self.env['account.report.line'].browse(lines_mapping.keys()):
+            report_lines_data[record.code] = lines_mapping[record.id]
+
+        # VP6a and VP6b values must be absolute values.
+        for code in ["VP6a", "VP6b"]:
+            if report_lines_data[code] and report_lines_data[code][0] == "-":
+                report_lines_data[code] = report_lines_data[code][1:]
+
+        return {
+            "supply_code": "IVP18",
+            "declarant_fiscal_code": options.get("declarant_fiscal_code"),
```

Comment:
It would be a programming error if it was not in the options
```suggestion
            "declarant_fiscal_code": options["declarant_fiscal_code"],
```